Fenrir/PackDeltaMethods.fs

```diff
@@ -14,46 +15,44 @@ let estimateSize (startSize: int) (off: int) (bytes: byte array): int =
             shift <- shift + 7
             int (elem % 128uy) <<< shift ||| acc) startSize bytes
 
-let byteToBits (num: byte) : byte[] =
-    let mutable number = num
-    Array.init 8 (fun _ ->
-        let result = number % 2uy
-        number <- number >>> 1
-        result)
+let byteToBits (num: byte) : BitArray =
+    BitArray([| num |])
 
 
-let bitsToInt (bits: byte[]) =
+
+let bitsToInt (bits: BitArray): int =
     let mutable st = 1
 
-    Array.foldBack (fun i acc ->
-        let result = acc + int i * st
+    seq { for b in bits do yield b }
```

Comment:
For reasons too convoluted to discuss here, you could try `Seq.cast<bool> bits`.